core/server/master/src/main/java/alluxio/master/backup/BackupWorkerRole.java

```diff
@@ -175,7 +177,7 @@ public BackupStatus getBackupStatus(BackupStatusPRequest statusPRequest) throws
     mBackupTimeoutTask = mTaskScheduler.schedule(() -> {
       LOG.info("Resuming journals as backup request hasn't been received.");
       enforceResumeJournals();
-    }, BACKUP_ABORT_AFTER_SUSPEND_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+    }, mBackupAbortSuspendTimeoutMs, TimeUnit.MILLISECONDS);
```

Comment:
Is there a concurrency issue here? since `mBackupTimeoutTask` is a shared state, could a `resume journal`  happen after the next suspend? so when we backup the journal is actually not in a suspended state?  I think we might have to stop the resume journal before we suspend?

Reply:
It can be a concern if a user sets the suspend timeout (default to 60s with this change) to be larger than the retry interval (10min) + the lock wait (30s). This is a constraint without this fix and will require changes on backup worker and journal system to make it reentrance safe. I don't think we need to worry about it unless we are looking for a much shorter retry interval or larger suspend timeout.

Reply:
discussed offline